lib/Models/BoxDrawing.ts

```diff
@@ -814,72 +829,72 @@ export default class BoxDrawing {
-        // tangential to the ellipsoid, giving an ellipsoidal point that is very
-        // far away. This causes large jumps in box position. To avoid this we
-        // angle the pick ray to the ellipsoid surface by first projecting the mouse
-        // points to the ellipsoid surface.
-
-        // The box origin
-        const origin = this.trs.translation;
-
-        // Box origin projected on the ellipsoid surface.
-        const surfacePoint = projectPointToSurface(origin, scratchSurfacePoint);
-        // Surface point in screen coordinates
-        const surfacePoint2d = scene.cartesianToCanvasCoordinates(
-          surfacePoint,
-          scratchSurfacePoint2d
+        // Move the box laterally on the globe while keeping the height
+        // steady. To do this we derive a new ellipsoid from the globe's
+        // ellipsoid by enlarging it so that the current box center falls on
+        // its surface. Then we find the intersection point of the camera ray
+        // through the latest mouse cursor position with this enlarged
+        // ellipsoid. The resulting point should be roughly the same height as
+        // the previous point and will also lie on the vector from the camera
+        // through the mouse cursor.
+
+        const cameraRay = scene.camera.getPickRay(
+          mouseMove.endPosition,
+          new Ray()
         );
-
-        if (!surfacePoint2d) {
-          // cartesianToCanvasCoordinates can unexpectedly return undefined.
+        if (!cameraRay) {
           return;
         }
 
-        // Floor the startPosition and endPosition above the ellipsoid.
-        const yDiff = mouseMove.endPosition.y - mouseMove.startPosition.y;
-        mouseMove.startPosition.y = surfacePoint2d.y;
-        mouseMove.endPosition.y = surfacePoint2d.y + yDiff;
-
-        // Fallback to simple ellipsoid pick when globe pick returns undefined
-        // (i.e there is no intersection of camera ray with globe tiles). This
-        // probably works only because ellipsoid might below the terrain so
-        // there is an intersection between the camera ray and ellipsoid.
-        // Although it could work, it doesn't truly fix the camera ray parallel
-        // to surface issue.
-        const previousPosition =
-          screenToGlobePosition(
-            scene,
-            mouseMove.startPosition,
-            scratchPreviousPosition
-          ) ??
-          scene.camera.pickEllipsoid(
-            mouseMove.startPosition,
-            undefined,
-            scratchPreviousPosition
-          );
-
-        const endPosition =
-          screenToGlobePosition(
-            scene,
-            mouseMove.endPosition,
-            scratchEndPosition
-          ) ??
-          scene.camera.pickEllipsoid(
-            mouseMove.endPosition,
-            undefined,
-            scratchEndPosition
-          );
-
-        if (!previousPosition || !endPosition) {
```

Comment:
We have improved the lateral movement of the box by keeping its height steady as we move it on the globe surface. 

To do this we derive a new ellipsoid from `globe.ellipsoid` which passes through the center of the box. This ellipsoid will roughly have the same shape as that of the `globe.ellipsoid`. Then we find the intersection point between the camera ray (ray from camera origin to location of users mouse cursor) and this new enlarged ellipsoid. This new point will lie on the ray and will have roughly the same height as the current box position. 

Because the new point lies on the camera ray, it makes it look like the box is following the mouse cursor.